.ci/jobsDSL/templates/mbp.groovy.j2

```diff
@@ -0,0 +1,171 @@
+            //discoverOtherRefsTrait {
+              // The pattern under /refs on the remote repository to discover, can contain a wildcard.
+              //ref('')
+              // Mapping for how the ref can be named in for example the @Library.
+              //nameMapping('')
+            //}
+            // Discovers branches on the repository.
+            // https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/BranchDiscoveryTrait.java#L55-L70
+            gitHubBranchDiscovery{
+              // Determines which branches are discovered.
+              strategyId(1)
+            }
+            // Discovers pull requests where the origin repository is the same as the target repository.
+            // https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/OriginPullRequestDiscoveryTrait.java#L57-L72
+            //gitHubPullRequestDiscovery {
+              // Determines how pull requests are discovered: Merging the pull request with the current target branch revision Discover each pull request once with the discovered revision corresponding to the result of merging with the current revision of the target branch.
+              //strategyId(1)
+            //}
+            //originPullRequestDiscoveryTrait {
+              //strategyId(3) //build both the head and merge refs
+            //}
+            // Discovers tags on the repository.
+            gitHubTagDiscovery()
+            // filers heads
+            headRegexFilter {
+              // A Java regular expression to restrict the names.
+              regex('(master|main|PR-.*|\\d+\\.x)')
+            }
+            // ignore push
+            //ignoreOnPushNotificationTrait()
+            // Defines a custom context label to be sent as part of Github Status notifications for this project.
+            notificationContextTrait {
+              // The text of the context label for Github status notifications.
+              contextLabel('{{ JOB_NAME }}')
+              // Appends the relevant suffix to the context label based on the build type.
+              typeSuffix(true)
+            }
+            wipeWorkspaceTrait()
+          }
+        }
+        buildStrategies {
+          buildChangeRequests {
+            // If the change request / pull request is a merge, there are two reasons for a revision change: The origin of the change request may have changed The target of the change request may ha
+            ignoreTargetOnlyChanges(false)
+            // Some sources can permit change request / pull request from external entities.
+            ignoreUntrustedChanges(true)
+          }
+          // Builds regular branches whenever a change is detected.
+          buildRegularBranches()
+          // Builds branches only if none of the sub strategies match.
+          //buildNamedBranches {
```

Comment:
Remove if unneeded?

Reply:
the template has some comment parts to help you to configure settings that are not in the default template, you only have to uncomment the setting and fill your configuration directly without having to search for it in the documentation.